File: src/main/java/view/BoardView.java
package view;

import domain.Board;
import org.jetbrains.annotations.NotNull;

class BoardView {
    private @NotNull TileBuffer mBuffer;

    BoardView(TileBuffer buffer) {
        mBuffer = buffer;
    }

    void render(@NotNull Board board) {
        char[][] grid = board.getGrid();

        mBuffer.initialize(grid.length);
        for (int y = 0; y < grid.length; y++) {
            char[] row = grid[y];

            for (int x = 0; x < row.length; x++) {
                mBuffer.set(x, y, new Tile(row[x]));
            }
        }
    }
}

File: Sources/Onitama/view/BoardView.swift
final class BoardView {
  private let mBuffer: TileBuffer

  // lifetime
  init(buffer: TileBuffer) {
    mBuffer = buffer
  }

  // command
  func render(board: Board) {
    let grid = board.grid

    mBuffer.initialize(size: grid.count)
    for y in 0..<grid.count {
      let row = grid[y]

      for x in 0..<grid.count {
        mBuffer.set(Point(x, y), tile: Tile(glyph: row[x]))
      }
    }
  }
}
